polyply/tests/test_lib_files.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Runs more elaborate integration tests
+"""
+
+from collections import defaultdict
+from pathlib import Path
+import shlex
+import subprocess
+import sys
+import numpy as np
+
+import pytest
+import vermouth
+from vermouth.forcefield import ForceField
+
+from polyply import TEST_DATA
+from vermouth.tests.helper_functions import find_in_path
+
+INTEGRATION_DATA = Path(TEST_DATA + '/library_tests')
+
+PATTERN = '{path}/{library}/{polymer}/polyply'
+
+POLYPLY = find_in_path(names=("polyply", ))
+
+def assert_equal_blocks(block1, block2):
+    """
+    Asserts that two blocks are equal to gain the pytest rich comparisons,
+    which is lost when doing `assert block1 == block2`
+    """
+    assert block1.name == block2.name
+    assert block1.nrexcl == block2.nrexcl
+    assert block1.force_field == block2.force_field  # Set to be equal
+    # Assert the order to be equal as well...
+    # assert list(block1.nodes) == list(block2.nodes)
+    # ... as the attributes
+
+    for node in block1.nodes:
+        for attr in ["atype", "charge"]:
+            assert block1.nodes[node][attr] == block2.nodes[node][attr]
```

Comment:
Why only these?

Reply:
they are the only ones relevant to the simulation; maybe could check the mass as well. The reason is some reference itps come straight from 3rd parties and they may differ in any other aspect except atom-type and charge.

Reply:
Add as comment